src/main/java/org/c02e/jpgpj/Encryptor.java

```diff
@@ -408,13 +411,68 @@ public void encrypt(InputStream plaintext, OutputStream ciphertext)
      * @throws PassphraseException if an incorrect passphrase was supplied
      * for one of the signing keys.
      */
-    public void encrypt(InputStream plaintext, OutputStream ciphertext,
-    FileMetadata meta) throws IOException, PGPException {
+    public void encrypt(
+        InputStream plaintext, OutputStream ciphertext, FileMetadata meta)
+            throws IOException, PGPException {
+        if (meta == null) {
+            meta = new FileMetadata();
+        }
+
+        try (OutputStream targetStream = prepareCiphertextOutputStream(ciphertext, meta)) {
+            // copy plaintext bytes into encryption pipeline
+            copy(plaintext, targetStream, null /* signer is inside the target */, meta);
+        }
+    }
+
+    public OutputStream prepareCiphertextOutputStream(File ciphertext)
+            throws IOException, PGPException {
+        // delete old output file
+        if (ciphertext.delete()) {
+            log.debug("prepareCiphertextOutputStream - deleted {}", ciphertext);
+        }
+
+        FileMetadata meta = new FileMetadata(ciphertext);
+        OutputStream fileStream = null;
+        try {
+            fileStream = new FileOutputStream(ciphertext);
+            OutputStream wrapper = prepareCiphertextOutputStream(fileStream, meta);
+            fileStream = null;  // avoid auto-close at finally clause
+            return wrapper;
+        } catch(Exception e) {
+            // delete output file if anything went wrong
+            if (fileStream != null) {
+                ciphertext.delete();
+            }
+            throw e;
+        } finally {
+            if (fileStream != null) {
+                fileStream.close();
+            }
+        }
+    }
+
+    /**
+     * Prepares an {@link OutputStream} - so that everything written to it
+     * is encrypted+compressed+signed - according to the encryptor's configuration
+     *
+     * @param ciphertext The original {@link OutputStream} into which the
+     * encryption results are to be written. <B>Note:</B> the stream will
+     * not be closed when the returned wrapper is closed
+     * @param meta The {@link FileMetadata} - if {@code null} an ad-hoc
+     * empty instance is used
```

Comment:
I'd suggest clarifying that this is for the original plaintext file's metadata -- something like: "The original plaintext file's {@link FileMetadata} if available - ..."

Reply:
Roger that...